src/trade-queue.js

```diff
@@ -0,0 +1,74 @@
+const TradeQueue = function () {
+    const {
+        setIntervalAsync,
+        clearIntervalAsync,
+    } = require("set-interval-async/dynamic")
+    const colors = require("colors")
+
+    const queue = []
+    const interval = 250
+    let setIntervalAsyncHandle
+
+    const addToQueue = (task) => {
+        queue.push(task)
+    }
+
+    const startQueue = () => {
+        setIntervalAsyncHandle = setIntervalAsync(
+            () => processQueue(),
+            interval
+        )
+    }
+
+    const stopQueue = async () => {
+        await clearIntervalAsync(setIntervalAsyncHandle)
+    }
+
+    const processQueue = async () => {
+        if (queue.length === 0) {
```

Comment:
Not sure what you're asking here. Tasks can be added to the queue at any point via the addToQueue function, and it will loop through all tasks in the queue array at that point.

Reply:
Ah, that got lost in the tango... See it now, I'll fix that as well :)